.spelling

```diff
@@ -1,15 +1,3 @@
-# markdown-spellcheck spelling configuration file
-# Format - lines beginning with # are comments
-# global dictionary is at the start, file overrides afterwards
-# one word per line, to define a file override use ' - filename'
-# followed by file specific exceptions, one per line
-# filename is relative to this configuration file
```

Comment:
Would be nice to retain these comments at the top of the file.

Also, even though we don't use file overrides currently, I think this sorting will mess those entries up. If we don't ever plan to use file overrides, I suppose the auto sort is a good thing.

Reply:
Could that be better served in the README potentially? Though it could also be done with a bit of shell script foo.

Reply:
> Would be nice to retain these comments at the top of the file.

Fixed.

> Also, even though we don't use file overrides currently, I think this sorting will mess those entries up.

I agree. However, I don't see any current examples, and not sure about its usage...

Reply:
>  I'll be happy to revise the script if we have any cases.

Even though we don't have any file overrides currently, I think it would be good to support it in the script so that it won't break if someone ever adds one.

I think the script just needs to `sed` for the first line in the file that matches ` - filename`, then only sort the lines up to that line and append the remaining lines without sorting. WDYT?

Reply:
He added a grep filter that does something similar but ignores all comments. LGTM but this is more infra so wanted to defer to you or Eric.

Reply:
> He added a grep filter that does something similar but ignores all comments.

Yes, I saw that. That's great. I'm suggesting he also handle the second problem (file overrides), probably pretty easy to do in his script using `sed`.

Reply:
Thinking about this some more, there are still 2 problems:

1. If we add any comments further down in the file (i.e., comment in front of some specific word, for example), it will get moved to the top of the file.
2. Any file overrides will not work.

Since comments don't really work, and the current comments are describing things (comments and file overrides) that don't work anyway, if we want to go with this auto sort feature, I think we need to remove all the comments (like it was before), and just use the basic list of words only, i.e., no comments or file overrides in the file.

@hanxiaop I suggest you revert your latest commit, since it doesn't really help, and then otherwise LGTM.

Reply:
@frankbu Done, thanks!